docs/pages/database-access/troubleshooting.mdx

```diff
@@ -39,3 +39,102 @@ details and examples:
 - [Redis Cluster](./guides/redis-cluster.mdx#step-46-set-up-mutual-tls)
 
 After the new certificate is issued, update your database to make it take effect.
+
+## Access to db denied
+
+Attempts to connect to the database fail with an error message
+similar to: **"access to db denied"**.
+
+**Solution:** Configure the database instance and Teleport role-based access control (RBAC) to allow the user 
+access.
+
+Check that the database user and database name (aka schema) you are trying to access via `tsh` exist in the database instance.
+When referring to "database name" or `db_names`, this is not the same as the `name` field in the `db_service` section of your Database Service configuration;
+this is referring to database names or schemas within a particular database instance.
+
+Once your database instance is configured, Teleport RBAC must be configured to allow access.
+Assign the Teleport user a role which **allows** `db_users`, `db_names`, and `db_labels` matching the 
+database user, database name, and Teleport static or dynamic labels for the database. Additionally, check that 
+the user does not have role(s) which **deny** the corresponding `db_users`, `db_names`, or `db_labels`.
+
+<Admonition type="warning">
+  Database names are only enforced for PostgreSQL and MongoDB databases.
+  When connecting to a PostgreSQL or MongoDB instance, be sure to specify `--db-name`:
+  ```sh
+  $ tsh db connect --db-user=exampleuser --db-name=exampledb
+  ```
+</Admonition>
+
+For example, the following Teleport user "alice@example.com"
```

Comment:
```suggestion
For example, the following `user` and `role` resources allow the Teleport user `alice@example.com` to
```
- Using code style so this doesn't show up as a link in the rendered paragraph
- Link this paragraph to the resources below more explicitly